docs/source/docs/registry.md

```diff
@@ -1,32 +1,108 @@
+- `value` is the registered class, e.g. `UNet` corresponding to `"unet"`.
+
+A global variable `REGISTRY = Registry()` is defined to provide a central control of all
+classes. The supported categories and the registered classes are resumed in the
+[registered classes](registered_classes.html) page.
+
+### Register a class
+
+To register a class into `REGISTRY`, it is recommended to use `register` as a
+**decorator**. Consider the `UNet` class for backbone as an example.
+
+```python
+from deepreg.registry import REGISTRY
+
+
+@REGISTRY.register(category="backbone_class", name="unet")
+class UNet:
+    """UNet-style backbone."""
+```
+
+The decorator automatically registers the class upon import. To ensure that this class
+is registered when `import deepreg` is called, this class and related parent modules
+need to be imported in the `__init__.py` files.
+
+For the purpose of code simplicity, a specific register function is defined for each
+category. For instance, we can use `register_backbone` for `UNet`:
+
+```python
+from deepreg.registry import REGISTRY
+
+
+@REGISTRY.register_backbone(name="unet")
+class UNet:
+    """UNet-style backbone."""
+```
+
+### Instantiate a class
+
+To instantiate a registered example in `REGISTRY`, we call `build_from_config` , which
+allows creating a class instance from a config directly. The config should be a
+dictionary containing the key `name` and other required keys for the class. Please check
+the [configuration documentation](configuration.html) and the docstring of the related
+classes for detailed configuration requirements. The path of the registered classes are
+resumed in the [registered classes](registered_classes.html) page.
+
+For instance, to instantiate a UNet class,
+
+```python
+from deepreg.registry import REGISTRY
+
+config = dict(name="unet",
+              image_size=(16,16,16),
+              out_channels=3,
+              num_channel_initial=2,
+              out_kernel_initializer="he_normal",
+              out_activation="")
+unet = REGISTRY.build_from_config(category="backbone_class", config=config)
+```
+
+where `kwargs` represents other required arguments.
+
+Similarly, for the purpose of code simplicity, a specific build function is defined for
+each category. For instance, we can use `build_backbone` for `UNet`:
+
+```python
+from deepreg.registry import REGISTRY
+
+
+config = dict(name="unet",
+              image_size=(16,16,16),
+              out_channels=3,
+              num_channel_initial=2,
+              out_kernel_initializer="he_normal",
+              out_activation="")
+unet = REGISTRY.build_backbone(config=config)
+```
+
+## Example usages
+
+To further explain how to use `REGISTRY` for using customized classes, detailed examples
+are provided for the following categories:
 
 - backbone
-- loss
+
+### Custom backbone
+
+To register a custom backbone class, the steps are as follows
+
+1. Inherit the `BackboneInterface` and implement a custom backbone class.
```

Comment:
- better to use "subclassing" instead of "inherit" to be consistent with other pytorch/keras language?
- Is the class name 'BackboneInterface' confusing? as your category is just 'backbone_class', it is not obvious what the rest of the class names are, "ModelInterface" and "LossInterface"? better to be just 'Backbone', 'Loss', 'Model'....

Reply:
@YipengHu ok

Reply:
done